src/main/java/org/mongojack/internal/util/SerializationUtils.java

```diff
@@ -195,14 +194,18 @@ private static Object serializeQueryCondition(
             return serializedConditions;
         } else {
             CompoundQueryCondition compound = (CompoundQueryCondition) condition;
-            if (!key.startsWith("$")) {
-                serializer = findQuerySerializer(false, key, serializerProvider, serializer);
```

Comment:
Can you add this to the DocumentSerializationUtils as well? I think ideally the portion of the library dealing with Mongo 2.X API should be deprecated.

Reply:
Done